torch/autograd/_functions/initializers.py

```diff
@@ -0,0 +1,20 @@
+from ..function import InplaceFunction
+
+
+class Zero(InplaceFunction):
+
+    @staticmethod
+    def forward(ctx, i, inplace=False):
+        if inplace:
+            ctx.mark_dirty(i)
+            result = i.zero_()
+        else:
+            result = i.new(i.size()).zero_()
+
+        ctx.save_for_backward(result)
+        return result
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        result, = ctx.saved_variables
+        return result.new([0]).expand_as_(result)
```

Comment:
correct me if i'm wrong, but aren't you returning a `Tensor` here? (also isn't `result` a Variable which doesn't have `new`?). This function isn't `once_differentiable` so i think backward stuff is all variables.

Reply:
@soumith is right, but `result.new([0]).expand_as(result)` is a bad idea. I doubt it will be any faster than `result.new(1).zero_().expand_as(result)`, but unlike the second version it is a GPU sync point!

Reply:
Have changed back to `result.new(1).zero_().expand_as(result)` and wrapped in `Variable` as @soumith is correct the backwards stuff is all variables.